edge/pkg/edged/cadvisor/cadvisor_linux.go

```diff
@@ -0,0 +1,93 @@
+// +build cgo,linux
+
+/*
+Copyright 2015 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cadvisor
+
+import (
+	"github.com/google/cadvisor/events"
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	cadvisorapi2 "github.com/google/cadvisor/info/v2"
+	"k8s.io/kubernetes/pkg/kubelet/cadvisor"
+)
+
+type cadvisorClient struct {
+	rootPath string
+}
+
+// New creates new cadvisor client
+func New(rootPath string) (cadvisor.Interface, error) {
+	return &cadvisorClient{
+		rootPath: rootPath,
+	}, nil
+}
+
+func (cadvisorClient) Start() error {
+	return nil
+}
+
+func (cadvisorClient) DockerContainer(name string, req *cadvisorapi.ContainerInfoRequest) (cadvisorapi.ContainerInfo, error) {
+	return cadvisorapi.ContainerInfo{}, nil
+}
+
+func (cadvisorClient) ContainerInfo(name string, req *cadvisorapi.ContainerInfoRequest) (*cadvisorapi.ContainerInfo, error) {
+	return nil, nil
+}
+
+func (cadvisorClient) ContainerInfoV2(name string, options cadvisorapi2.RequestOptions) (map[string]cadvisorapi2.ContainerInfo, error) {
+	return nil, nil
+}
+
+func (cadvisorClient) SubcontainerInfo(name string, req *cadvisorapi.ContainerInfoRequest) (map[string]*cadvisorapi.ContainerInfo, error) {
+	return nil, nil
+}
+
+func (cadvisorClient) MachineInfo() (*cadvisorapi.MachineInfo, error) {
+	return &cadvisorapi.MachineInfo{
+		4,
```

Comment:
how are these hard-coded values used ? can we retrieve them from the system ?

Reply:
we don't want to introduce cadvisor server into edge_core, so we implement the function MachineInfo by hard code.

Reply:
done